assets/app/scripts/directives/oscKeyValues.js

```diff
@@ -139,7 +144,8 @@ angular.module("openshiftConsole")
         deletePolicy: "@",
         readonlyKeys: "@",
         keyValidationTooltip: "@",
-        valueValidationTooltip: "@"
+        valueValidationTooltip: "@",
+        canBeEmpty: "@",
```

Comment:
If you use `canBeEmpty: "="`, you shouldn't need the special boolean handling for `canBeEmpty` in `compile` below.